libs/docs/src/docs/2.2.1-alpha/overriding-pod-and-container-attributes.md

```diff
@@ -0,0 +1,95 @@
+---
+title: Extend Kubernetes resources for container components
+description: Extend Kubernetes resources for container components through attributes
+---
+
+This section describes how you can extend kubernetes resources for container components through `pod-overrides` and `container-overrides` attributes.
+
+These attributes can be defined at Devfile level and component level. [Strategic Merge Patch](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-api-machinery/strategic-merge-patch.md#basic-patch-format) strategy is used to merge the attributes defined at both level.
+
+
+## container-overrides
+
+`container-overrides` attributes allows you to override properties of a container in a pod spec such as `securityContext` and `resources`. However, it restricts from overriding properties such as `image`, `name`, `ports`, `env`, `volumesMounts`, `command`, and `args`.
+
+### Procedure
+1. Specify "container-overrides" at component level.
+    ```yaml {% title="Specify container-overrides to override security context for container at component level" filename="devfile.yaml" %}
+    ----
```

Comment:
We can remove the dashes and also add `schemaVersion: <version>` for the devfile examples

Reply:
What will be the schemaVersion here? 2.2.1 or 2.2.0? Also, do I need to backport this doc to 2.2.0 and 2.1.0?

Reply:
> We can remove the dashes
All the dashes? Top and bottom ones?